pkg/proxy/iptables/proxier.go

```diff
@@ -591,6 +593,47 @@ func (proxier *Proxier) OnEndpointSlicesSynced() {
 	proxier.syncProxyRules()
 }
 
+// OnNodeAdd is called whenever creation of new node object
+// is observed.
+func (proxier *Proxier) OnNodeAdd(node *v1.Node) {
+	if node.Name != proxier.hostname {
+		klog.Errorf("Received a watch event for a node %s that doesn't match the current node %v", node.Name, proxier.hostname)
+		return
+	}
+	proxier.mu.Lock()
+	proxier.nodeLabels = node.Labels
+	proxier.mu.Unlock()
+}
+
+// OnNodeUpdate is called whenever modification of an existing
+// node object is observed.
+func (proxier *Proxier) OnNodeUpdate(oldNode, node *v1.Node) {
+	if node.Name != proxier.hostname {
+		klog.Errorf("Received a watch event for a node %s that doesn't match the current node %v", node.Name, proxier.hostname)
+		return
+	}
+	proxier.mu.Lock()
+	proxier.nodeLabels = node.Labels
```

Comment:
I think we need to explicitly trigger a sync (in all 3 handlers), don't we?  Otherwise this will be deferred until the next service or endpoint change.  Or am I missing some connection?  E.g.


```
lock
old = proxier.nodeLabels
proxier.nodeLabels = new
unlock

if old != new  { 
  proxier.syncProxyRules()
}
```

Reply:
Yes, sorry It's my mistake. It seems we still need DeepEqual to compare the old and new label, but we can compare length at first to avoid traversing when labels been added or removed.